### ISSUE TITLE
Export selected entries/groups to new database

### ISSUE THREAD

**Issue by droidmonkey** [keepassxreboot/keepassxc#1940]
## Feature Request

Be able to export selected entries to create a new database from a master database. Exported entries may be kept or removed from the master database upon export.

Requested by a customer.

## Possible Solution

Add to the CLI and/or make a GUI tool

**Comment by ThatMandem** [comment]
Yes. Please.

**Comment by jbrice** [comment]
That would be _so_ useful!

**Comment by Pentaphon** [comment]
I have always wanted this. It would be a good way to make DBs for less devices like a Windows PC that is always online. You can export everything except email, bank, shopping and government passwords so that if you get pwned, they will get only that DB with trivial logins like forums, entertainment, etc , not the one with the most sensitive logins.

**Comment by droidmonkey** [comment]
Technically you can do this now rather easily using KeeShare

**Comment by spacemanspiff2007** [comment]
Currently copy paste between different databases does not work. A possibility to copy/move certain entries to a different database would be highly apreciated!

**Comment by droidmonkey** [comment]
Also allow export to CSV and HTML for printing

**Comment by Asmodan123** [comment]
A very nice extension of this "Export Entries to an other DB" feature would be to persist the target DB on the selected entries to repeat the export periodically. This would be helpful to create a primary DB which contains all password, which are changed/updated periodically. Than the export to the Sub-DB can be repeated easily.

**Comment by droidmonkey** [comment]
That feature is already available using KeeShare. Read the user guide for more information.

**Comment by Sebi2020** [comment]
> That feature is already available using KeeShare. Read the user guide for more information.

This is really not the same as the requested feature...

**Comment by droidmonkey** [comment]
I am responding to the comment above, thanks

**Comment by thyxNY** [comment]
W will wait for export to html/csv some entries. It is nice to see this feature in 2.8.0 ;-)

For me best is:
- export search result to html/new kdbx
- export highlighted items to html/new kdbx

I will wait ;-)

**Comment by genepine** [comment]
Hello
I wrote a blog about how to implement this function in the operation method.
https://pinefield.asia/portforio/keepassxc-export-selected-groups-to-new-database/

**Comment by Avihais12344** [comment]
This option is available on the original Keepass software.

**Comment by gehaxelt** [comment]
Please implement this :+1:

**Comment by lewart** [comment]
> This is really not the same as the requested feature...

If you need to export some selected entries, it means they have something in common.
It may be a good idea to organize them in a group.
A KeeShare feature, which is implemented since 2.4, forces you to organize your database.
Is that the difference from request above, youre talking about?

**Comment by ttomasini** [comment]
> > This is really not the same as the requested feature...
> 
> If you need to export some selected entries, it means they have something in common. It may be a good idea to organize them in a group. A KeeShare feature, which is implemented since 2.4, forces you to organize your database. Is that the difference from request above, youre talking about?

@lewart thank you very much for your proposal but I think this feature request actually contains two separate use cases.

Only one would be maybe covered by your idea:
* sharing only one or several particular subsets of a database
* exporting the raw data of entries/groups into another data format esp. in order to further process the entries (e.g. for external programs or printing out)

Maybe it is a good idea to distinguish between these two cases in separate issues as well.

**Comment by droidmonkey** [comment]
Both of those use cases are covered under this issue and would be part of the same code contribution. No need for another issue.

**Comment by Sebi2020** [comment]
@lewart 
> If you need to export some selected entries, it means they have something in common.
> It may be a good idea to organize them in a group.
> A KeeShare feature, which is implemented since 2.4, forces you to organize your database.
> Is that the difference from request above, youre talking about?

Yes, but I would agree with you about organizing entries which have something in common as a group.  But if we're talking about exports you maybe have more than one way to categorize items. Maybe you want to export some passwords on a regular basis from different groups, but don't want to export all passwords of a group. I think entry tags would be much more suitable for this use case as they allow you to categorize an item in more than one way. For example you could have a category "Wifi passwords" and a separate category "Exportable passwords". If you combine this with a feature "Export tag entries" this would give users much more flexibility.

**Comment by lewart** [comment]
> I think entry tags would be much more suitable for this use case as they allow you to categorize an item in more than one way. 
> If you combine this with a feature "Export tag entries" this would give users much more flexibility.

I agree. Sharing not only by group, but also by tag, would be realy useful.

**Comment by feefladder** [comment]
+1 for the tagging system!
Also, for people that didn't find it in the documentation, but did find this issue, here is the link to keeshare documentation:
https://keepassxc.org/docs/KeePassXC_UserGuide#_database_sharing_with_keeshare
(The table of contents is really long, I was confused by that at first, but docs look great in general)

**Comment by arkanoid87** [comment]
6 years, really?

**Comment by droidmonkey** [comment]
Absent writing code for this, here is your solution (takes about 30 seconds):

1. Unlock database you want to export from
2. Create new database, nothing special required
3. Drag group / entries from source database, hold CTRL, drag to new Database, drop. A copy of the entries will be made
4. Export new Database to CSV
5. Close new Database, don't save changes

**Comment by arkanoid87** [comment]
thanks! I also managed to workaround the issue by moving to database, save, copy the dest db file, move back, delete intermediate db

dragging with CTRL was not obvious. How about right click drag + contextual menu move/copy?

**Comment by droidmonkey** [comment]
certainly can be investigated, holding CTRL while dragging to make a copy is relatively standard. Works in Windows Explorer as well ;-)

**Comment by genepine** [comment]
> thanks! I also managed to workaround the issue by moving to database, save, copy the dest db file, move back, delete intermediate db
> 
> dragging with CTRL was not obvious. How about right click drag + contextual menu move/copy?

Use in combination with the CTRL key to select multiple entries within a folder. 
For example, if you have entry ABC, you want to select only A and C.

1. Hold down the CTRL key and select entries A and C in the folder on database A
2. Drag to the tab of the database B you want to copy
3. Once the database B is switched, drop it into the folder you want to copy.
4. Save database B , don't save database A

**Comment by noraj** [comment]
I often have to share some entries with a colleague working on the same mission as me. The more straightforward way for the user would to have the existing export menu (global) available on contextual menu (right click) when a group or a selection of entries is selected. Export as CSV/HTML/XML as well as "Export as a new database (.kdbx)".

I would be way quicker than the [current workaround](https://github.com/keepassxreboot/keepassxc/issues/1940#issuecomment-2025262591).

**Comment by ki4cjt-0597** [comment]
It would be very useful to be able to export an individual entry, multiple selected entries, and an entire group of entries instead of only being able to export the entire database.

Having the ability to export to an HTML file, or PDF with properly formatted blocks or tables would be awesome.

I recall having that functionality in the standard keepass app but cannot seem to find any way to do it in KeepassXC. If I'm overlooking something (and I tried searching it out too but didn't find anything) then please point me to what I need to know....otherwise if that could be a feature added in the near future I'm sure myself and many others would love it.

One use-case that comes to mind... If I have passwords stored in my master database, separated out by groups, and lets say all those groups are specific clients that I'm storing my login credentials to various services that I support for said clients...save them to individual client groups. Then if I need to hand over a client's set of passwords/info, being able to selectively export said entries, or their entire group would be very useful.

**Comment by Kurgan-** [comment]
I can't believe that after 7 years the only solution is to create a new database. The code to convert data to HTML or whatever is already there, is it so hard to make it work on a selection of data instead of the whole db?

**Comment by droidmonkey** [comment]
Your code contributions are welcome
